great_expectations/zep/context.py

```diff
@@ -0,0 +1,149 @@
+from __future__ import annotations
+
+import logging
+from pprint import pformat as pf
+from typing import TYPE_CHECKING, Callable, Dict, List, Optional
+
+from great_expectations.util import camel_to_snake
+from great_expectations.zep.type_lookup import TypeLookup
+
+if TYPE_CHECKING:
+    from great_expectations.zep.interfaces import Datasource
+
+SourceFactoryFn = Callable[..., "Datasource"]
+
+LOGGER = logging.getLogger(__name__)
+
+
+def _remove_suffix(s: str, suffix: str) -> str:
+    # NOTE: str.remove_suffix() added in python 3.9
+    if s.endswith(suffix):
+        s = s[: -len(suffix)]
+    return s
+
+
+def _get_simplified_name_from_type(
+    t: type, suffix_to_remove: Optional[str] = None
+) -> str:
+    result = camel_to_snake(t.__name__)
+    if suffix_to_remove:
+        return _remove_suffix(result, suffix_to_remove)
+    return result
+
+
+class _SourceFactories:
+    """
+    Contains a collection of datasource factory methods in the format `.add_<TYPE_NAME>()`
+
+    Contains a `.type_lookup` dict-like two way mapping between previously registered `Datasource`
+    or `DataAsset` types and a simplified name for those types.
+    """
+
+    type_lookup = TypeLookup()
```

Comment:
I find `type_lookup` hard to reason about because it does a lot of work. It's a bidirectionally dictionary that stores both information about datasources and data assets. It would be easier for me if we broke this up. I also think we will need to namespace data assets to datasources. How do you feel about making a `data_asset_type_lookup` and a `datasource_type_lookup`?

Reply:
I was thinking we might need to do something exactly like that, but I was punting on that until I got to actually integrating the config loading/deserialization stuff.